Sources/Common/Communication/Event.swift

```diff
@@ -24,71 +24,120 @@ public extension EventRepresentable {
+        self.storageId = storageId
         self.name = name
         self.params = params
     }
 }
 
 public struct ResetEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
 
-    public init(params: [String: String] = [:]) {
+    public init(storageId: String = UUID().uuidString, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.params = params
     }
 }
 
 public struct TrackMetricEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
     public let deliveryID: String
     public let event: String
     public let deviceToken: String
 
-    public init(deliveryID: String, event: String, deviceToken: String, params: [String: String] = [:]) {
+    public init(storageId: String = UUID().uuidString, deliveryID: String, event: String, deviceToken: String, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.params = params
         self.deliveryID = deliveryID
         self.event = event
         self.deviceToken = deviceToken
     }
 }
 
+public struct TrackInAppMetricEvent: EventRepresentable {
+    public var storageId: String
+    public var params: [String: String] = [:]
+    public let deliveryID: String
+    public let event: String
+
+    public init(storageId: String = UUID().uuidString, deliveryID: String, event: String, params: [String: String] = [:]) {
+        self.storageId = storageId
+        self.params = params
+        self.deliveryID = deliveryID
+        self.event = event
+    }
+}
+
 public struct RegisterDeviceTokenEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
     public let token: String
 
-    public init(token: String, params: [String: String] = [:]) {
+    public init(storageId: String = UUID().uuidString, token: String, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.token = token
         self.params = params
     }
 }
 
 public struct DeleteDeviceTokenEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
 
-    public init(params: [String: String] = [:]) {
+    public init(storageId: String = UUID().uuidString, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.params = params
     }
 }
 
 public struct NewSubscriptionEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
     public let subscribedEventType: String
 
-    init<E: EventRepresentable>(subscribedEventType: E.Type, params: [String: String] = [:]) {
+    init<E: EventRepresentable>(storageId: String = UUID().uuidString, subscribedEventType: E.Type, params: [String: String] = [:]) {
+        self.storageId = storageId
         self.subscribedEventType = E.key
     }
 }
```

Comment:
this comment applies for all structs above
* You don't need constructors to define default values
* `var` will fire back badly

Reply:
Added response in previous comment